src/serialization.rs

```diff
@@ -261,37 +260,42 @@ mod test {
         (proof, vk)
     }
 
-    #[test]
-    fn test_proof_vk_serialization_tweedledee() {
-        let (proof, vk) = get_circuit_vk::<Tweedledee, Tweedledum>();
-        let ser_proof = bincode::serialize(&proof).unwrap();
-        let ser_vk = bincode::serialize(&vk).unwrap();
-
-        let der_proof = bincode::deserialize(&ser_proof).unwrap();
-        let der_vk: VerificationKey<Tweedledee> = bincode::deserialize(&ser_vk).unwrap();
-
-        assert_eq!(proof, der_proof);
-        assert_eq!(vk.c_constants, der_vk.c_constants);
-        assert_eq!(vk.c_s_sigmas, der_vk.c_s_sigmas);
-        assert_eq!(vk.degree, der_vk.degree);
-        assert_eq!(vk.num_public_inputs, der_vk.num_public_inputs);
-        assert_eq!(vk.security_bits, der_vk.security_bits);
-    }
 
-    #[test]
-    fn test_proof_vk_serialization_tweedledum() {
-        let (proof, vk) = get_circuit_vk::<Tweedledum, Tweedledee>();
-        let ser_proof = bincode::serialize(&proof).unwrap();
-        let ser_vk = bincode::serialize(&vk).unwrap();
-
-        let der_proof = bincode::deserialize(&ser_proof).unwrap();
-        let der_vk: VerificationKey<Tweedledum> = bincode::deserialize(&ser_vk).unwrap();
-
-        assert_eq!(proof, der_proof);
-        assert_eq!(vk.c_constants, der_vk.c_constants);
-        assert_eq!(vk.c_s_sigmas, der_vk.c_s_sigmas);
-        assert_eq!(vk.degree, der_vk.degree);
-        assert_eq!(vk.num_public_inputs, der_vk.num_public_inputs);
-        assert_eq!(vk.security_bits, der_vk.security_bits);
+    macro_rules! test_proof_vk_serialization {
+        ($curve:ty, $inner_curve:ty, $test_name:ident) => {
+            #[test]
+            fn $test_name() -> Result<()> {
+                let (proof, vk) = get_circuit_vk::<$curve, $inner_curve>();
+                let ser_proof = serde_cbor::to_vec(&proof)?;
+                let ser_vk = serde_cbor::to_vec(&vk)?;
+                let vk_no_fft = { let mut vk_no_fft = vk.clone(); vk_no_fft.clear_fft_precomputation(); vk_no_fft };
```

Comment:
Maybe slightly cleaner to do e.g. `let vk_no_fft = VerificationKey { fft_precomputation: None, ..vk }`